### PR TITLE
204 add enrollment

### PR DESCRIPTION
**Pull request by mmazanec22**

### CODE REVIEW

File: client/src/containers/AddChild/AddChild.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  useEffect(() => {
+    // On initial load, create child
+    if (child || childId || !accessToken) return;
+
+    if (!organization) {
+      throw Error('Cannot create child without organization');
+    }
+    apiPost(
+      'children',
+      { firstName: '', lastName: '', organization },
```
Comment (jlhogan) [inline client/src/containers/AddChild/AddChild.tsx:130]: this seems weird to me, is it necessary? we need to create the child before the user has supplied any info?

File: client/src/containers/CheckData/CheckData.tsx [hunk 2]
```diff
@@ -28,6 +28,10 @@ const CheckData: React.FC = () => {
     }
   }, [reportId, accessToken]);
 
+  const organization =
```
Comment (jlhogan) [inline client/src/containers/CheckData/CheckData.tsx:31]: I think we'd want to get all the orgs from the authenticated user, and if more than one the user has to pick, otherwise just that org the user has access to  ?

File: client/src/containers/EditRecord/EditRecord.tsx [hunk 3]
```diff
@@ -60,14 +61,8 @@ const EditRecord: React.FC = () => {
...
   // Get reporting periods (needed to update enrollments with fundings)
-  const [reportingPeriods, setReportingPeriods] = useState<ReportingPeriod[]>(
-    []
-  );
-  useEffect(() => {
-    apiGet('reporting-periods', { accessToken }).then((_reportingPeriods) =>
-      setReportingPeriods(_reportingPeriods || [])
-    );
-  }, [accessToken]);
+  // TODO: we should probably use context rather than making lots of network requests
```
Comment (jlhogan) [inline client/src/containers/EditRecord/EditRecord.tsx:64]: well like -- we should have a data cache

File: client/src/containers/EditRecord/Forms/ChildInfo/Form.tsx [hunk 4]
```diff
@@ -94,3 +91,5 @@ export const ChildInfoForm: React.FC<ChildInfoFormProps> = ({
     </Form>
   );
 };
+
+export default ChildInfoForm;
```
Comment (jlhogan) [inline client/src/containers/EditRecord/Forms/ChildInfo/Form.tsx:95]: why?

File: client/src/containers/EditRecord/Forms/EnrollmentFunding/Fields/Funding/NewFunding.tsx [hunk 5]
```diff
@@ -73,6 +75,7 @@ export const FundingField = <T extends ChangeFunding | ChangeEnrollment>({
       id="funding-source"
       name="funding-source"
       legend="Funding source options"
+      showLegend
```
Comment (jlhogan) [inline client/src/containers/EditRecord/Forms/EnrollmentFunding/Fields/Funding/NewFunding.tsx:78]: this should always be true?

File: client/src/containers/EditRecord/Forms/FamilyIncome/Fields/DeterminationCard.tsx [hunk 6]
```diff
@@ -8,8 +8,8 @@ import {
   ExpandCard,
   Pencil,
 } from '@ctoec/component-library';
-import { currencyFormatter } from '../../../../../utils/formatters';
 import { IncomeDetermination } from '../../../../../shared/models';
+import currencyFormatter from '../../../../../utils/formatters/currency';
```
Comment (jlhogan) [inline client/src/containers/EditRecord/Forms/FamilyIncome/Fields/DeterminationCard.tsx:12]: why this change?

File: client/src/containers/EditRecord/Forms/FamilyIncome/Form.tsx [hunk 7]
```diff
@@ -24,14 +13,18 @@ export type IncomeFormProps = {
...
+export const FamilyIncomeForm: React.FC<EditFormProps> = ({
+  child,
+  onSuccess,
 }) => {
   const [showRedeterminationForm, setShowRedeterminationForm] = useState(false);
   const [currentIsNew, setCurrentIsNew] = useState(false);
 
+  if (!child) return <></>;
+
+  const determinations = child.family.incomeDeterminations;
```
Comment (jlhogan) [inline client/src/containers/EditRecord/Forms/FamilyIncome/Form.tsx:25]: should this be `child.family?.incomeDeterminations` ? same with below.
or will family always exist at this point?

**Review by jlhogan — APPROVED** [review]

File: client/src/containers/AddChild/AddChild.tsx (hunk 1 above)
Comment (mmazanec22) [reply client/src/containers/AddChild/AddChild.tsx:130]: This seemed like the easiest way to reuse forms, so that child info doesn't need to know whether it is creating a child on save.  If we prefer to stick this in child info and give it a create new child param or something that works too, also open to other suggestions

Comment (jlhogan) [reply client/src/containers/AddChild/AddChild.tsx:130]: ahhh i see, that makes sense. my only worry is that we create the data and then network error/user gets a phone call/etc and they abandon the data in this state, then it's very hard to display to the user in a way that makes sense -- it'd be a  totally empty row. but we can cross that bridge when/if that happens, and can always update to move PUT/POST decisioning in childinfo form if necessary.

File: client/src/containers/CheckData/CheckData.tsx (hunk 2 above)
Comment (mmazanec22) [reply client/src/containers/CheckData/CheckData.tsx:31]: @jlhogan since this is going to require some design decisions, ok to punt to a new ticket in backlog?

File: client/src/containers/EditRecord/Forms/FamilyIncome/Fields/DeterminationCard.tsx (hunk 6 above)
Comment (mmazanec22) [reply client/src/containers/EditRecord/Forms/FamilyIncome/Fields/DeterminationCard.tsx:12]: Got an error because it was being exported by default but needs to be named-- changing it to be a named export

File: client/src/containers/EditRecord/Forms/EnrollmentFunding/Fields/Funding/NewFunding.tsx (hunk 5 above)
Comment (mmazanec22) [reply client/src/containers/EditRecord/Forms/EnrollmentFunding/Fields/Funding/NewFunding.tsx:78]: I think because of our header/legend issues we are usually not showing it.  Resolving that should be combined with the other header fixes probably

File: client/src/containers/EditRecord/Forms/FamilyIncome/Form.tsx (hunk 7 above)
Comment (mmazanec22) [reply client/src/containers/EditRecord/Forms/FamilyIncome/Form.tsx:25]: I think family is required for a child to be created?  And we're compensating for the potentially undefined state of dets below this so I'll just move that up here for clarity
